feflow/tests/test_hybrid_topology.py

```diff
@@ -0,0 +1,143 @@
+"""
+Module to implement the basic unit testing for the hybrid topology implementation.
+Specifically, regarding the HybridTopologyFactory object.
+More oriented to testing code functionality than science correctness.
+"""
+import pytest
+
+from feflow.utils.hybrid_topology import HybridTopologyFactory
+from feflow.tests.utils import extract_htf_data
+
+from openmm import unit as omm_unit
+from perses.tests import utils as perses_utils
+
+
+@pytest.fixture(scope="module")
+def standard_system_generator():
+    """
+    Fixture to create a standard/default system generator based on commonly used options for
+    force fields, temperature and pressure. That is, amber forcefields for proteins, openff 2.1.0
+    for small molecules, 1 bar pressure and 300k temperature.
+
+    Returns
+    -------
+    generator: openmmforcefields.generators.SystemGenerator
+        Dictionary with the key, value pairs to directly use when creating an instance of
+        openmmforcefields.generators.SystemGenerator.
+    """
+    from openmm.app import NoCutoff, PME
+    from openmm import MonteCarloBarostat
+    from openmmforcefields.generators import SystemGenerator
+
+    sys_gen_config = {}
+    sys_gen_config["forcefields"] = ["amber/ff14SB.xml",
+                                     "amber/tip3p_standard.xml",
```

Comment:
the best place to deal with environment virtual sites is here - I can probably spend the hour tomorrow porting over a set of tests that will deal with this (it's mostly atom counting and ensuring coordinates & particle properties are set properly).